### PR TITLE
Update stealing script to support multiple provinces

### PR DESCRIPTION
**Pull request by charleyc-dr**
Also fix a few bugs; in particular, don't write multiple copies of a new item.

### CODE REVIEW

File: steal.lic [hunk 1]
```diff
@@ -104,7 +115,7 @@ class Steal
   end
 
   def do_buff(buffs, settings)
-    buffs['spells'].each { |spell| cast_spell(spell, settings) }
+    buffs['spells'].each { |spell| buff(spell, settings) }
```
Comment (jandersson) [inline steal.lic]: the buff method is deprecated and going to be removed #2337

Comment (jandersson) [reply steal.lic]: OK, now it is removed :P

Comment (charleyc-dr) [reply steal.lic]: Oops, ok, reverting this!  Was an unintentional reversion during merge.

**Comment by charleyc-dr** [comment]
Per discussion in Slack, going to do a bit more work to refine the "province" notion.  I've got this done but need to test it before I send it out.

**Comment by charleyc-dr** [comment]
Ok, I've split things up by town and added the option to list the towns you're willing to steal from.

**Comment by rpherbig** [comment]
Does the default `stealing_towns` (in base.yaml) match how stealing is currently behaving?

File: data/base-stealing.yaml [hunk 2]
```diff
@@ -4648,29 +4646,21 @@
...
 - id: '409'
   item: bone backer
   item_in: in crate
   pawnable: false
-  province: zoluren
+  town: Crossing
   room: 8864
   trivial_max: 1750
   trivial_min: 741
-- id: '410'
```
Comment (rpherbig) [inline data/base-stealing.yaml]: Why was this removed?

File: profiles/base.yaml [hunk 3]
```diff
@@ -269,7 +269,13 @@ steal_past_mindlock: false
 stealing_buffs:
 npc_stealing_attempt_count: 0
 dont_steal_list:
-
+stealing_towns:
```
Comment (rpherbig) [inline profiles/base.yaml]: Do these default stealing_towns match how stealing is currently behaving?

File: steal.lic [hunk 4]
```diff
@@ -134,6 +137,19 @@ class Steal
     move('go door')
   end
 
+  def walk_to_bin
+    case @hometown
+    when 'Crossing'
```
Comment (rpherbig) [inline steal.lic]: This feels like something that should be in `base-town.yaml`. What do you think?

File: steal.lic [hunk 5]
```diff
@@ -134,6 +137,19 @@ class Steal
     move('go door')
   end
 
+  def walk_to_bin
```
Comment (rpherbig) [inline steal.lic]: If this is a predicate, could you name it `walk_to_bin?` ?

File: steal.lic [hunk 6]
```diff
@@ -335,36 +372,36 @@ class Steal
               'you decide to wait a while before trying again',
               'Roundtime')
     when /this trivial theft/
-      update_target(target, 'trivial')
+      update_target(target, 'trivial') if count == 1
```
Comment (rpherbig) [inline steal.lic]: Why did you add the conditional here?

File: steal.lic [hunk 7]
```diff
@@ -20,7 +20,13 @@ class Steal
 
     @stealing_options = get_data('stealing').stealing_options.reject { |target| settings.dont_steal_list.include?(target['id'].to_i) }
     @stealing_bag = settings.stealing_bag
+    @hometown = settings.hometown
+    @stealing_towns = settings.stealing_towns
     @bin_stolen = settings.bin_stolen
+    unless %w[Crossing].include?(@hometown) || !@bin_stolen
```
Comment (jandersson) [inline steal.lic]: It would  be nice to couple the towns with the existence of a bin room. That way the script doesn't have to be updated for any new bin data

File: data/base-stealing.yaml (hunk 2 above)
Comment (charleyc-dr) [reply data/base-stealing.yaml]: It's a dupe of item 394.

File: profiles/base.yaml (hunk 3 above)
Comment (charleyc-dr) [reply profiles/base.yaml]: Yes, these are exactly the towns where items are already in the db.

File: steal.lic (hunk 6 above)
Comment (charleyc-dr) [reply steal.lic]: These were being called multiple times per theft.  Usually that's harmless, but for a new item it was making N copies if you stole N times.

**Comment by charleyc-dr** [comment]
Ok, I've switched to using a bin location in base-town.yaml.
